File: Suma de columnas de una matriz/B1.java
package ejercicios;

import java.util.ArrayList;
import java.util.regex.Pattern;

public class Ejercicios {

    public static void main(String[] args) {
        String NumerosMatriz = "3 4 2 2 3 4 5 6 7 8 9 10 11 12";
        String x = "-1";
        String f = NumerosMatriz.substring(0,1);
        String c = NumerosMatriz.substring(2,3);
        NumerosMatriz = NumerosMatriz.substring(4,NumerosMatriz.length());
        
        String patron = "[ ]+";
        Pattern p1 = Pattern.compile(patron);
        String[] items = p1.split(NumerosMatriz);
        String n[] = items;
        ArrayList<String> Lista = new ArrayList<>();
        
            for (String s : items) {
               Lista.add(s);
            }
            
            
            int Matriz[][] = new int[Integer.parseInt(f)][Integer.parseInt(c)];
            for(int i = 0; i < Matriz.length; i++){
                for(int j = 0; j < Matriz[i].length; j++){
                    Matriz[i][j] = Integer.parseInt(Lista.get(0));
                    Lista.remove(0);
                }
            }
            
            int con = 0;
            int columna = 0;
            int suma = 0;
             
           
             for(int i = 0; i < Integer.parseInt(c); i++){
                for(int j = 0; j < Integer.parseInt(f); j++){
                
                    con = con + Matriz[j][i];
                    if(con > suma){
                    suma = con;
                    columna = i;
                    
                }
                    
                }
                
             con = 0;    
            }
             
             System.out.println(columna + " " + suma);
    }

}

File: Suma de columnas de una matriz/B1.py
NumerosMatriz = "3 4 2 2 3 4 5 6 7 8 9 10 11 12"
x = "-1"
f = NumerosMatriz[0:1]
c = NumerosMatriz[2:3]

NumerosMatriz = NumerosMatriz[4:len(NumerosMatriz)]

Lista = NumerosMatriz.split()

Matriz = []
for i in range(0, int(f)):
	Matriz.append([])
	for j in range(0, int(c)):
		Matriz[i].append(Lista[0])
		del Lista[0]


con = 0
columna = 0
suma = 0

for fila in Matriz:
	print(fila)

for i in range(0, int(c)):
	for j in range(0, int(f)):
		con = con + int(Matriz[j][i])
		if con > suma :
			suma = con
			columna = i
	con = 0

print(str(columna) + " " + str(suma))